File: solutions/921/921-kir3i.java
class Solution {
    public int openCnt = 0;
    public int closeCnt = 0;
    public int minAddToMakeValid(String S) {
        for(int i=0; i<S.length(); i++) {
            if (S.charAt(i) == '(')
                openCnt++;
            else if(S.charAt(i) == ')') {
                if (openCnt > 0)
                    openCnt--;
                else
                    closeCnt++;
            }
        }
        return openCnt + closeCnt;
    }
}

File: solutions/921/921-kir3i.py
class Solution:
    def minAddToMakeValid(self, S):
        openCnt = 0
        closeCnt = 0
        for c in S:
            if c == '(':
                openCnt += 1
            elif c == ')':
                if openCnt > 0:
                    openCnt -= 1
                else:
                    closeCnt += 1
        return openCnt+closeCnt
